src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java

```diff
@@ -142,11 +158,18 @@ public JoinProgramResponse joinProgram(
 
   public ListProgramsResponse listPrograms(Predicate<ProgramEntity> predicate) {
     val programEntities =
-        programService.listPrograms().stream().filter(predicate).collect(toList());
+        programService.listPrograms().stream()
+            // Only show active programs
+            .filter(p -> p.getActive().booleanValue())
+            .filter(predicate)
+            .collect(toList());
     return programConverter.programEntitiesToListProgramsResponse(programEntities);
   }
 
   public ListUsersResponse listUsers(String programShortName) {
+    // Fetching the program first will throw an error if it is not active or doesnt exist
+    //   stopping the ego requests for a program that was never initialized
+    programService.getProgram(programShortName);
```

Comment:
this is a little weird, you use the get to check the program exists??

Reply:
yeup, hence the comments lol